test/nl/surf/eduhub_rio_mapper/e2e_helper.clj

```diff
@@ -0,0 +1,349 @@
+
+(defn- print-api-message
+  "Print boxed API request and response."
+  [{{:keys [method url]}  :req
+    {:keys [status body]} :res}]
+  (print-boxed
+      "API"
+    (println (str/upper-case (name method)) url status)
+    (when body
+      (print-json body))))
+
+(defn- print-rio-message
+  "Print boxed RIO request and response."
+  [{{:keys                [method url]
+     req-body             :body
+     {action :SOAPAction} :headers} :req
+    {res-body :body
+     :keys    [status]}             :res}]
+  (print-boxed
+      "RIO"
+    (println (str/upper-case method) url status)
+    (println "- action:" action)
+    (println "- request:\n")
+    (print-soap-body req-body)
+    (println)
+    (when (= http-status/ok status)
+      (println "- response:\n")
+      (print-soap-body res-body)
+      (println))))
+
+(defn- print-ooapi-message
+  "Print boxed OOAPI request and response."
+  [{{:keys [method url]}  :req
+    {:keys [status body]} :res}]
+  (print-boxed
+      "OOAPI"
+    (println (str/upper-case method) url status)
+    (println)
+    (when (= http-status/ok status)
+      (print-json-str body))))
+
+(defn- print-http-messages
+  "Print HTTP message as returned by API status."
+  [http-messages]
+  (when-let [{:keys [req] :as msg} (first http-messages)]
+    (if (and (= "post" (:method req))
+             (-> req :headers :SOAPAction))
+      (print-rio-message msg)
+      (print-ooapi-message msg))
+    (recur (next http-messages))))
+
+
+
+;; Defer running make-config so running some (other!) tests is still
+;; possible when environment incomplete.
+(def config (memoize #(cli/make-config)))
+
+(def base-url
+  (memoize
+   #(str "http://"
+         (-> (config) :api-config :host)
+         ":"
+         (-> (config) :api-config :port))))
+
+(defn- encode-base64
+  "Base64 bytes of given string."
+  [s]
+  (str/join (map char (.encode (Base64/getEncoder) ^bytes (.getBytes s)))))
+
+(def ^:private bearer-token
+  ;; A conext bearer token expires in 3600 seconds (1 hour), should be
+  ;; plenty of time to run all e2e tests..
+  (future
+    (let [{:keys [client-id
+                  client-secret
+                  token-endpoint]} env]
+      (-> {:method       :post
+           :url          token-endpoint
+           :content-type :x-www-form-urlencoded
+           :query-params {"grant_type" "client_credentials"
+                          "audience"   client-id}
+
+           :headers {"Authorization" (str "Basic "
+                                          (encode-base64 (str client-id
+                                                              ":"
+                                                              client-secret)))}
+           :as      :json}
+          (http/request)
+          (get-in [:body :access_token])))))
+
+(defn- api-path
+  "Returns path for given API action."
+  [action args]
+  (case action
+    :token
+    (let [[token] args]
+      (str "/status/" token))
```

Comment:
I'd prefer to use `:get-status` or just `:status` instead of `:token` -- we're not getting a token, we're getting a job status